resourcegroupstaggingapi/resourcegroupstaggingapi_test.go

```diff
@@ -0,0 +1,190 @@
+	},
+	{
+		resourceType: "elasticloadbalancing:targetgroup",
+		tags: []tag{
+			{
+				key:   "spinup:org",
+				value: "foobar",
+			},
+			{
+				key:   "spinup:spaceid",
+				value: "321",
+			},
+		},
+		arn: "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/testtg321/0987654321",
+	},
+}
+
+func (m *mockResourceGroupsTaggingAPIClient) GetResourcesWithContext(ctx context.Context, input *resourcegroupstaggingapi.GetResourcesInput, opts ...request.Option) (*resourcegroupstaggingapi.GetResourcesOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+
+	resourceList := []*resourcegroupstaggingapi.ResourceTagMapping{}
+	for _, r := range testResources {
+		if len(input.ResourceTypeFilters) > 0 {
+			var typeMatch bool
+			for _, t := range input.ResourceTypeFilters {
+				if aws.StringValue(t) == r.resourceType {
+					typeMatch = true
+					break
+				}
+			}
+
+			if !typeMatch {
+				continue
+			}
+		}
+
+		matches := true
+		for _, filter := range input.TagFilters {
+			innerMatch := func() bool {
+				m.t.Logf("processing tagfilter %+v", filter)
+				for _, rt := range r.tags {
+					if aws.StringValue(filter.Key) == rt.key {
+						m.t.Logf("tag keys match for %s (%s = %s)", r.arn, rt.key, aws.StringValue(filter.Key))
+						if len(filter.Values) == 0 {
+							m.t.Logf("appending %s to the list, keys match (%s = %s) and no value specified", r.arn, rt.key, aws.StringValue(filter.Key))
+							return true
+						}
+
+						for _, value := range aws.StringValueSlice(filter.Values) {
+							if value == rt.value {
+								m.t.Logf("appending %s to the list, keys match (%s = %s) and value matches (%s = %s)", r.arn, rt.key, aws.StringValue(filter.Key), value, rt.value)
+								return true
+							}
+						}
+					}
+				}
+				m.t.Logf("returning false for %s", r.arn)
+				return false
+			}()
+
+			if !innerMatch {
+				matches = false
+			}
+		}
+
+		if matches {
+			m.t.Logf("resource %s matches", r.arn)
+			resourceList = append(resourceList, &resourcegroupstaggingapi.ResourceTagMapping{
+				ResourceARN: aws.String(r.arn),
+			})
+		}
+	}
+
+	return &resourcegroupstaggingapi.GetResourcesOutput{
+		ResourceTagMappingList: resourceList,
+	}, nil
+}
+
+func TestGetResourcesWithTags(t *testing.T) {
+	r := ResourceGroupsTaggingAPI{Service: newmockResourceGroupsTaggingAPIClient(t, nil)}
+	filters := []*TagFilter{
+		{
+			Key:   "spinup:org",
+			Value: []string{"foobar"},
+		},
+		{
+			Key:   "spinup:spaceid",
+			Value: []string{"123"},
+		},
+	}
+	out, err := r.GetResourcesWithTags(context.TODO(), []string{"elasticloadbalancing:targetgroup"}, filters)
+	if err != nil {
+		t.Errorf("expected nil error, got %s", err)
+	}
+
+	expected := []string{"arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/testtg123/0987654321"}
+	if !reflect.DeepEqual(expected, out) {
+		t.Errorf("expected %+v, got %+v", expected, out)
+	}
+
+	out, err = r.GetResourcesWithTags(context.TODO(), []string{}, filters)
+	if err != nil {
```

Comment:
oh, I see empty types might actually be a valid option (that returns all types), should've looked at the test first 😄